libs/common-types/src/lib.rs

```diff
@@ -53,6 +56,40 @@ pub enum PoolRole<TrancheId = [u8; 16], Moment = u64> {
 	TrancheInvestor(TrancheId, Moment),
 }
 
+#[derive(Encode, Decode, Clone, Copy, PartialEq, Eq, TypeInfo, Debug)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub enum PermissionedCurrencyRole<Moment = u64> {
+	/// This role can hold & transfer tokens
+	Holder(Moment),
+	/// This role can add/remove holders
+	Manager,
+	/// This role can mint/burn tokens
+	Issuer,
+}
+
+#[derive(Encode, Decode, Clone, Copy, PartialEq, Eq, TypeInfo, Debug)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub enum Role<TrancheId = [u8; 16], Moment = u64> {
+	PoolRole(PoolRole<TrancheId, Moment>),
+	PermissionedCurrencyRole(PermissionedCurrencyRole<Moment>),
+}
```

Comment:
Could you leave a type-level comment explaining how this type is used overall and within its specific variants?

Reply:
thanks a lot @offerijns 👍